client/src/controllers/BlockController.ts

```diff
@@ -0,0 +1,83 @@
+import { Controller } from '@hotwired/stimulus';
+import { object } from 'prop-types';
+
+declare global {
+  interface Window {
+    initBlockWidget?: (id: string) => void; // Declare the initBlockWidget function as optional on the Window interface
+    telepath: any;
+  }
+}
+
+/**
+ * Adds the ability to unpack a Telepath object and render it on the controlled element.
+ *
+ * @example
+ * <div
+ * id="{id}"
+ * data-block
+ * data-controller="w-block"
+ * data-w-block-data-value="{block_json}"
+ * data-w-block-initial-value="{value_json}"
+ * data-w-block-error-value="{error_json}">
+ * </div>
+ */
+
+export class BlockController extends Controller<HTMLElement> {
+  /**
+   * initial - Initial value for the block
+   * data - Unpacked Telepath data
+   * error - Block error (optional)
+   */
+  static values = {
+    initial: { type: object, default: undefined },
+    data: { type: object, default: undefined },
+    error: { type: object, default: undefined },
+  };
```

Comment:
OK, so the current approach as is will not work correctly, we should be using `Object` not `object` as this is the Stimulus value declaration. 

Also, it appears that this code does not work as is, so you will need to use `initial` as an array it seems, but please double check.

Have a good read of https://stimulus.hotwired.dev/reference/values to understand what's going on here.

```suggestion
  static values = {
    initial: { type: Array, default: [] },
    error: { type: Object, default: {} },
    data: { type: Object, default: {} },
  };
```

Also the JSDoc should not be on the static property but in the TypeScript declarations below.